raiden-ts/src/config.ts

```diff
@@ -65,7 +65,7 @@ export const RaidenConfig = t.readonly(
       expiryFactor: t.number, // must be > 1.0
       httpTimeout: t.number,
       discoveryRoom: t.union([t.string, t.null]),
-      pfs: t.union([Address, t.string, t.null]),
+      pfs: t.union([t.readonlyArray(t.union([Address, t.string])), t.boolean]),
```

Comment:
First of all I like you general approach for this configuration option. But I would argue it could be improved and simplified at the same time. The name `pfs` is pretty unspecific, but the type behind it has a not trivial documentation. Without loosing any functionality, what do you think about making this just of type `t.readonlyArray(t.union([Address, t.string])`? Then we can somehow (sorry that I fail to come with a suggestion) rename the option that expresses that this list are manual/additional/exceptional services. And if this list is empty we just use the automated procedure.
What this approach does not configure is the case what to do if none of the configured services services is working. I would argue to just fail as the user explicitly wants to use these services (because of reasons?) and do not continue to auto-detect services. Having a second option to specify this behavior would be more precise or rather let the user decide instead of us. The downside here is just that we need to explain the co-interaction between these two options and which is has priority in which case. But I would argue that isn't worse what we have now for this single very complex option.
This are my thoughts/suggestions. What do you think?

PS: I think the following code somehow shows that the handling of such a complex single configuration options is quite cumbersome. Especially with such a flat name as the other parts of the code have no such easy access to the documentation of it.

Reply:
How would it be if the user wants to disable PFS?
I'm not a big fan of putting all these behaviors on this same config, but since they're somehow exclusive (false=disabled, true=auto, `(Address | string)[]`), I did it like that. An alternative would be to have an explicit list of `additionalServices`, then have the current option be an enum or a value which would go like `disabled`, `auto` (use any/best between additional+auto, additional are also notified) and `onlyAdditional` (notify everyone, use only one of the explicitly set value). It'd be more explicit, but I'm not sure the additional verbosity and complexity of two options are worth it instead of just allowing it be `false|true`.

Reply:
Hmm okay. I'm not 100% happy with how it is, but I like your suggestion. Though I don't see an obvious path to go for here. So if you don't feel the need to change then close this conversation and keep it as is. :+1:

Reply:
We discussed in a personal call and agreed on a solution which got now implemented in 4e438511579d247329de884fb4432f98f474f636